googleapis/src/main/java/io/grpc/googleapis/GoogleCloudToProdNameResolver.java

```diff
@@ -267,7 +279,17 @@ private boolean queryIpv6SupportMetadata(String url) throws IOException {
     HttpURLConnection con = null;
     try {
       con = httpConnectionProvider.createConnection(url);
-      return con.getResponseCode() == 200;
+      if (con.getResponseCode() != 200 ) {
+        return false;
+      }
+      InputStream inputStream = con.getInputStream();
+      if (inputStream == null) {
+        return false;
+      }
+      try (Reader reader = new InputStreamReader(inputStream, Charsets.UTF_8)) {
+        String respBody = CharStreams.toString(reader);
+        return !respBody.isEmpty();
+      }
```

Comment:
I know this is not a high performance part of the code, but if for some reason the response had a lot of data in the body you would read all of that into memory for no reason. How about just reading the first byte from the input stream and if you get nothing you know it's empty?

Reply:
It would be really strange to have a long response, but you are right that it is much cleaner to just do the simple read.